### ISSUE TITLE
Switching to the alternate screen stops refreshing the terminal in 3.1.8

### ISSUE THREAD

**Issue by Farmbuyer** [mintty/mintty#1015]
Using 3.1.8 under 64-bit Cygwin (DLL 3.1.5) on Windows 10 Enterprise.

Starting terminal-based Vim 8.2.0486 correctly "flips" to the alternate screen with the editor contents, but then stops redrawing the cursor or anywhere the cursor has been.  (I realize this is hard to follow; it's equally hard to describe.)  The cursor starts in the standard top-left corner, but stays as a solid black box rather than solid white, or blinking, or anything else.  As I move the cursor around, none of the characters it passes over are redrawn after it leaves.  There's just a solid black "worm trail" of where the cursor has been.

Exiting the editor does not properly restore/refresh the original screen; it just changes back to the shell prompt right in the middle of the contents of the file I was editing.  A 'reset' will clear the screen and restore a normal blinking-white box cursor on a normal black background, but rerunning vim just starts the weirdness all over again.

Something similar happens with any other terminal/curses program.  Running 'hg commit -i' for an interactive commit, for example, has lines with a +/- expand/collapse action.  Expanding a line into more text incorrectly draws that text over top of lower lines, instead of moving them down.  Exiting the program doesn't flip the screen back, only prints the shell prompt over top of the screen contents until a tset/reset is ran.

Downgrading to 3.1.6 restores normal editing.  (I didn't test 3.1.7; since I had upgraded *from* 3.1.6 to 3.1.8, I figured I'd just retreat to a known state.)  The only other Cygwin packages I had updated lately besides Mintty were some perl libraries, so neither the Vim nor Mercurial binaries have changed.

I don't have any particularly weird mintty settings - as far as I know!  The TERM is set to xterm-256color.  I'm happy to report any other mintty or DLL or Windows 10 settings or values; I just don't know which ones might be affecting this behavior.

**Comment by mintty** [comment]
Please make a screen log (mintty -l .logfile). You can switch on or off logging from the extended context menu (Ctrl+right-click). Stop it when the effect is visible, in order not to include any resetting control.
Then try to reproduce the issue with `cat .logfile`. Upload the log here.
If you have X installed, you could also try the issue with another terminal.

**Comment by Farmbuyer** [comment]
Okay, next time I get a chance to halt the ongoing stuff, I'll re-upgrade and capture a logfile.  Might be a few days.

**Comment by Farmbuyer** [comment]
Argh.  Our IT dept has pushed Windows patches, including a bunch with terminal- and pty-related labels.  After re-installing 3.1.8 and launching with logging, I can no longer reproduce the issue.

I cannot roll back the Windows patches to try and duplicate it, and even if we found it there's presumably not much that can be done on the mintty side, if it's native libraries screwing up.  Might as well close this.

God, I hate Windows.

**Comment by mintty** [comment]
This hardly explains the phenomenon as mintty does not use any special Windows API for switching its terminal screen, just plain normal text rendering as for all output. But who knows...
